session/promise/storage.go

```diff
@@ -146,6 +147,45 @@ func (s *Storage) GetAllKnownIssuers() []identity.Identity {
 	return res
 }
 
+var errNoPromiseForConsumer = errors.New("no promise for consumer")
+
+// FindPromiseForConsumer returns the last known promise for the issuer/consumer combo
+// It checks if any promises match, and if they do checks to see if any later promises have been cleared.FindPromiseForConsumer
+func (s *Storage) FindPromiseForConsumer(issuerID, consumerID identity.Identity) (StoredPromise, error) {
+	s.Lock()
+	defer s.Unlock()
+	promises, err := s.getAllPromisesForIssuer(issuerID)
+	if err != nil {
+		return StoredPromise{}, err
+	}
+
+	// sort by sequenceID, descending
+	sort.Slice(promises, func(i, j int) bool {
```

Comment:
We probably can just call `sortPromisesDesc` here.

Reply:
fixd